plaid-stl/src/lib.rs

```diff
@@ -78,9 +78,68 @@ pub mod messages;
 
 //pub use ::plaid::LogSource;
 
+#[macro_export]
+macro_rules! setup_non_allocating_panic_mgmt {
+    () => {
+        use std::fmt::Write;
+
+        // A simple stack-allocated buffer with fixed size
+        struct NoAllocBuffer<'a> {
+            buf: &'a mut [u8],
+            pos: usize,
+        }
+
+        impl<'a> NoAllocBuffer<'a> {
+            fn new(buf: &'a mut [u8]) -> Self {
+                NoAllocBuffer { buf, pos: 0 }
+            }
+        }
+
+        impl<'a> std::fmt::Write for NoAllocBuffer<'a> {
```

Comment:
For clarity: I asked ChatGPT to come up with a non-allocating version of the code I had before.
I tested this code but we still end up with "This is probably an OOM", which is the default message when everything else fails.